### PR TITLE
SUPESC-313: Fixed the volume prices for different price types.

### PR DESCRIPTION
**Pull request by komendant-dev**
Branch: backport/supesc-313/price-product-2.15.0
Ticket: https://spryker.atlassian.net/browse/SUPESC-313
Target Version: 2.15.0

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   PriceProduct               | minor                 |                       |

#### Release Notes

{skip}

-----------------------------------------

##### Change log

Improvements
 
- Introduced `CurrentProductPriceTransfer.priceDataByPriceType` transfer property.
- Introduced `PriceProductConfig::PRICE_DATA_BY_PRICE_TYPE`.
- Adjusted `PriceProductFacade::findPricesBySkuGroupedForCurrentStore()` and `PriceProductFacade::groupPriceProductCollection()` so they group price data also by a price type.
- Adjusted `PriceProductClient::resolveProductPriceTransferByPriceProductFilter()` and `PriceProductClient::resolveTransfer()` so now them also populate `CurrentProductPriceTransfer::priceDataByPriceType` transfer property.

Deprecations

- Deprecated `CurrentProductPriceTransfer.priceData` transfer property.

### CODE REVIEW

File: src/Spryker/Shared/PriceProduct/Transfer/price_product.transfer.xml [hunk 1]
```diff
@@ -90,6 +90,7 @@
         <property name="quantity" type="int" />
         <property name="sumPrice" type="int" />
         <property name="priceData" type="string"/>
```
Comment (abitskil) [inline src/Spryker/Shared/PriceProduct/Transfer/price_product.transfer.xml]: * Don't see a deprecation of the `priceData` property.

File: src/Spryker/Zed/PriceProduct/Business/Model/PriceGrouper.php [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<?php
+4<?php
```
Comment (abitskil) [inline src/Spryker/Zed/PriceProduct/Business/Model/PriceGrouper.php]: Remove `4`.

File: tests/SprykerTest/Zed/PriceProduct/Business/PriceProductFacadeTest.php [hunk 3]
```diff
@@ -47,6 +47,18 @@ class PriceProductFacadeTest extends Unit
     public const EUR_ISO_CODE = 'EUR';
     public const USD_ISO_CODE = 'USD';
 
+    protected const PRICE_TYPE_ORIGINAL = 'ORIGINAL';
```
Comment (abitskil) [inline tests/SprykerTest/Zed/PriceProduct/Business/PriceProductFacadeTest.php]: * Undo the changes related to tests from the backport.

Comment (v-n7k) [reply tests/SprykerTest/Zed/PriceProduct/Business/PriceProductFacadeTest.php]: Done

File: src/Spryker/Zed/PriceProduct/Business/Model/PriceGrouper.php (hunk 2 above)
Comment (v-n7k) [reply src/Spryker/Zed/PriceProduct/Business/Model/PriceGrouper.php]: Fixed, sorry!

**Review by tamasnyulas — APPROVED** [review]
